syntax = "proto3";

import "messages.proto";
import "compatibility.proto";

service ClientConnector {

    // This should always be run on app/client start.
    rpc CheckCompatibility(SemVer) returns (CompatibilityInfo) {}

    rpc SendMessage(Message) returns (SendMessageResponse) {}
    rpc SubscribeToMessages(SubscribeToMessagesParams) returns (stream Message) {}
}

message SubscribeToMessagesParams {}
